26.0-rc3/master segmentation fault in `increase_proc_bin_sz`

**Describe the bug**
Segmentation fault when handling an HTTP request (scraping RabbitMQ Prometheus metrics). I don't have a simpler test case at this point, so unfortunately reproduction steps involve starting RabbitMQ. I will see if I can isolate the root cause (most things, including the Prometheus endpoint, work fine).

**To Reproduce**
```
git clone https://github.com/rabbitmq/rabbitmq-server.git
git checkout otp26-compatibility
make run-broker

curl 'http://localhost:15692/metrics/detailed?family=vhost_status'
```

**Expected behavior**
no segfault :)

**Affected versions**
26.0-rc3, `master-bb3848576b`; consistently reproducible in multiple environments (Linux, macOS)

**Additional context**
Full coredump: https://github.com/mkuratczyk/files/raw/main/coredump-master.gz
beam.smp: https://github.com/mkuratczyk/files/raw/main/beam.smp.gz

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000055d88a2afff2 in increase_proc_bin_sz (p=0x7f0ff4f8bd50, pb=0x1f3149, new_size=10) at beam/erl_bits.c:1382
1382        if (new_size > pb->size) {
[Current thread is 1 (Thread 0x7f0ff785c640 (LWP 1282289))]
(gdb) bt
#0  0x000055d88a2afff2 in increase_proc_bin_sz (p=0x7f0ff4f8bd50, pb=0x1f3149, new_size=10) at beam/erl_bits.c:1382
#1  0x000055d88a2b0fa5 in erts_bs_private_append_checked (p=0x7f0ff4f8bd50, bin=139706699840482, build_size_in_bits=72, unit=8)
    at beam/erl_bits.c:1699
#2  0x00007f0ffab27438 in ?? ()
#3  0x0000000000000000 in ?? ()
```

I have not been able to reproduce neither on macOS nor Linux.

On my Mac, trying to run `make run-broker`, I get first:

```
erlang.mk:26: Please upgrade to GNU Make 4 or later: https://erlang.mk/guide/installation.html
```

Is it mandatory to use Make 4?

Later I get:

```
** (Mix) Cannot compile dependency :stdout_formatter because it isn't available, please ensure the dependency is at "/Users/bjorng/git/rabbitmq-server/deps/stdout_formatter"
make[1]: *** [escript/rabbitmqctl] Error 1
make: *** [install-cli-escripts] Error 2
```

If I run just `make` from a clean repository, it ends with:

```
Error: No Makefile to build dependency /Users/bjorng/git/rabbitmq-server/deps/syslog.
```

On my Linux system, when I run `make run-broker` I get:

```
make[2]: Entering directory '/home/uabbgus/git/rabbitmq-server/deps/thoas'
make[2]: Entering directory '/home/uabbgus/git/rabbitmq-server/deps/thoas'
Makefile:21: reltool_util: No such file or directory
make[2]: *** No rule to make target 'reltool_util'.  Stop.
make[2]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/thoas'
make[1]: *** [erlang.mk:4512: deps] Error 2
make[1]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/rabbit_common'
make: *** [erlang.mk:4512: deps] Error 2
make: *** Waiting for unfinished jobs....
Makefile:21: reltool_util: No such file or directory
make[2]: *** No rule to make target 'reltool_util'.  Stop.
make[2]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/thoas'
make[1]: *** [erlang.mk:8301: /home/uabbgus/git/rabbitmq-server/.erlang.mk/recursive-deps-list.log] Error 2
make[1]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/rabbit_common'
make: *** [erlang.mk:8300: /home/uabbgus/git/rabbitmq-server/.erlang.mk/recursive-deps-list.log] Error 2
```
When I run just `make`, I get:

```
/home/uabbgus/git/rabbitmq-server/deps/rabbitmq_cli/test/diagnostics/log_tail_command_test.exs

         |
118 118  |    Enum.map(logs, fn log ->
119 119  |      case log do
120     -|        '<stdout>' -> :ok
    120 +|        ~c"<stdout>" -> :ok
121 121  |        _ -> File.write(log, "")
122 122  |      end
         |


make[2]: *** [Makefile:103: escript/rabbitmqctl] Error 1
make[2]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/rabbitmq_cli'
make[1]: *** [../../erlang.mk:4512: deps] Error 2
make[1]: Leaving directory '/home/uabbgus/git/rabbitmq-server/deps/rabbit'
make: *** [erlang.mk:4512: deps] Error 2
```

--- a/lib/compiler/src/beam_ssa_alias.erl
+++ b/lib/compiler/src/beam_ssa_alias.erl
@@ -874,12 +874,12 @@ aa_make_fun(Dst, Callee=#b_local{name=#b_literal{}},
     %% and ensure that the value is only used once, even if the
     %% argument dies at this location.
     %%
-    %% We also update the call info for the callee in the #aas{} to
-    %% reflect the aliased status for the arguments coming from the
-    %% environment.
+    %% We also assume that all arguments are aliased because someone
+    %% could have stolen the function through tracing and called it
+    %% with unexpected arguments, which may be aliased.
     SS = aa_set_aliased([Dst|Env0], SS0),
     #{ Callee := Status0 } = Info0,
-    Status = aa_merge_env(reverse(Status0), [aliased || _ <- Env0], []),
+    Status = [aliased || _ <- Status0],
     #{ Callee := PrevStatus } = Info0,
     Info = Info0#{ Callee := Status },
     Repeats = case PrevStatus =/= Status of
@@ -892,13 +892,6 @@ aa_make_fun(Dst, Callee=#b_local{name=#b_literal{}},
               end,
     AAS = AAS0#aas{call_args=Info,repeats=Repeats},
     {SS, AAS}.
-
-aa_merge_env([_|Args], [E|Env], Acc) ->
-    aa_merge_env(Args, Env, [E|Acc]);
-aa_merge_env([Arg|Args], [], Acc) ->
-    aa_merge_env(Args, [], [Arg|Acc]);
-aa_merge_env([], [], Acc) ->
-    Acc.
 
 aa_breadth_first(Funs, FuncDb) ->
     IsExported = fun (F) ->

--- a/lib/compiler/test/bs_construct_SUITE.erl
+++ b/lib/compiler/test/bs_construct_SUITE.erl
@@ -31,7 +31,7 @@
 	 two/1,test1/1,fail/1,float_bin/1,in_guard/1,in_catch/1,
 	 nasty_literals/1,coerce_to_float/1,side_effect/1,
 	 opt/1,otp_7556/1,float_arith/1,otp_8054/1,
-         strings/1,bad_size/1]).
+         strings/1,bad_size/1,private_append/1]).
 
 -include_lib("common_test/include/ct.hrl").
 
@@ -47,7 +47,7 @@ groups() ->
       [verify_highest_opcode,
        two,test1,fail,float_bin,in_guard,in_catch,
        nasty_literals,side_effect,opt,otp_7556,float_arith,
-       otp_8054,strings,bad_size]}].
+       otp_8054,strings,bad_size,private_append]}].
 
 
 init_per_suite(Config) ->
@@ -754,3 +754,19 @@ bad_binary_size2() ->
 
 bad_binary_size3(Bin) ->
     <<Bin:all/binary>>.
+
+%% GH-7121: Alias analysis would not mark fun arguments as aliased, fooling
+%% the beam_ssa_private_append pass.
+private_append(_Config) ->
+    <<"alpha=\"alpha\",beta=\"beta\"">> =
+        private_append_1(#{ <<"alpha">> => <<"alpha">>,
+                            <<"beta">> => <<"beta">> }),
+
+    ok.
+
+private_append_1(M) when is_map(M) ->
+    maps:fold(fun (K, V, Acc = <<>>) ->
+                        <<Acc/binary, K/binary, "=\"", V/binary, "\"">>;
+                    (K, V, Acc) ->
+                        <<Acc/binary, ",", K/binary, "=\"", V/binary, "\"">>
+                end, <<>>, M).

